signinapp/admin/users.py

```diff
@@ -59,10 +66,19 @@ def new_user():
             name=form.name.data,
             password=form.password.data,
             approved=form.approved.data,
-            role=Role.query.get(form.role.data)
+            role=Role.query.get(form.role.data),
+            preferred_name=form.preferred_name.data,
+            phone_number=form.phone_number.data,
+            email=form.email.data,
+            address=form.address.data,
+            tshirt_size=ShirtSizes[form.tshirt_size.data],
         )
         if form.subteam.data:
             user.subteam = Subteam.query.get(form.subteam.data)
+
+        # If no preferred name was provided default to their name
+        if not form.preferred_name.data:
```

Comment:
We talked about it offline, but for posterity - `preferred_name` should be nullable, and an accessor (`@hybrid_property`) should be made `display_name` that uses preferred if it exists or falls back to `name`